dts/bindings/flash_controller/zephyr,disk-flash.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Emulates a flash using a disk
+
+compatible: "zephyr,disk-flash"
+
+include: base.yaml
+
+properties:
+  size:
+    type: int
+    description: flash capacity in bits
+    required: true
+  disk-name:
+    type: string
+    description: the name of the underlying disk to be used
+    required: true
+  disk-sector-size:
+    type: int
+    description: the sector size in byte of the underlying disk to be used
+    required: true
+  disk-offset:
+    type: int
+    description: >
+      offset on the disk in bytes. If the offset is 10 the 10th byte will
+      be the first one used.
+    default: 0
```

Comment:
True I could get the sector size at runtime which I already do but it is needed at compile-time in order to properly size the buffer `buf_##index`. Do you have a better proposal how to do this?

Why is disk-offset problematic? Or what exactly is problematic here?

Reply:
The information is already there, there must not be an other property to describe the same, please take a look at DT_INST_PROP_BY_PHANDLE, ping me if you need help with that.

Reply:
@jfischer-no I would appreciate some help here. I don't understand where this would already be defined since this is the offset for the driver in order to enable multiple partitions/flash instances on the same underlying disk.
 
 This enables the user to define multiple flash partitions on a disk by specifying their offset and size.

Reply:
I mean disk-sector-size, I can try it out in the next few days and post it here.

Reply:
@kica-z For the disk-name and disk-sector-size, see https://github.com/jfischer-no/zephyr/commit/0ad08a83ac43cc26209334453bfe38535355930a.patch, you can squash it into your commit an drop my signed-off.

Reply:
@jfischer-no Very nice patch thanks for that 👍

Unfortunately it does not seem to work with `sdmmc-disk` as far as I could tell since no `sector-size` and `disk-name` devicetree properties are present. If that should be the case I don't think we should include the patch since it would narrow down the possible list of underlying disks.

I may however have made a mistake during testing which is unrelated to the changes in the patch. @jfischer-no what do you think? Should it work with the mmc disk node?

# Setup:

## Devicetree
```
&spi3 {
  status = "okay";
  cs-gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
  pinctrl-0 = <&spi3_pinctrl>;
  pinctrl-names = "default";

  sdhc0: sdhc@0 {
      compatible = "zephyr,sdhc-spi-slot";
      reg = <0>;
      status = "okay";
      pwr-gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
      mmc0: mmc0 {                                                        <----- This was used as phandle
          compatible = "zephyr,sdmmc-disk";
          status = "okay";
      };
      spi-max-frequency = <8000000>;
  };
};
```

## Error
`error: 'DT_N_S_soc_S_peripheral_50000000_S_spi_b000_S_sdhc_0_S_mmc0_P_disk_name' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_spi_b000_S_sdhc_0_S_mmc0_P_status'?`

Reply:
> @jfischer-no Very nice patch thanks for that 👍
> 
> Unfortunately it does not seem to work with `sdmmc-disk` as far as I could tell since no `sector-size` and `disk-name` devicetree properties are present. If that should be the case I don't think we should include the patch since it would narrow down the possible list of underlying disks.

That is unfortunately an issue in SDMMC driver, I spoke about this at ZDS2023, multi-instance support is broken for sdmmc drivers, someone needs to fix it.

Reply:
Oh ok, that is unfortunate. Is there an issue open for this already?
I would propose not to include the patch until this has been fixed since sdmmc is the disk driver thats most relevant for this PR.